src/sentry/services/organization/provisioning.py

```diff
@@ -26,60 +28,97 @@ class OrganizationProvisioningException(Exception):
 
 
 class OrganizationProvisioningService:
+    def _validate_or_default_region(self, region_name: Optional[str]):
+        silo_mode = SiloMode.get_current_mode()
+        if region_name is None and silo_mode == SiloMode.CONTROL:
+            raise OrganizationProvisioningException(
+                "A region name must be provided when provisioning an organization from the Control Silo"
+            )
+
+        if region_name is None:
+            region_name = get_local_region().name
```

Comment:
Should we protect against the `region_name` not matching the current region's name?

Reply:
Yeah that's a good call, I'll update this to prevent that case.